.github/CODEOWNERS

```diff
@@ -29,7 +29,8 @@ qiskit/utils/         @Qiskit/terra-core @manoelmarques @woodsp-ibm
 providers/            @Qiskit/terra-core @jyu00
 quantum_info/         @Qiskit/terra-core @ikkoham
 qpy/                  @Qiskit/terra-core
-pulse/                @Qiskit/terra-core @eggerdj  @wshanks
+pulse/                @Qiskit/terra-core @eggerdj @wshanks
+synthesis/            @Qiskit/terra-core @alexanderivrii @ShellyGarion
```

Comment:
Yeah, `synthesis/` implicitly means `**/synthesis` (with `globstar` semantics).  For most cases, the reason is because we needed `{qiskit,test/python}/synthesis` to be matched, but here, it'll pull in a bit more than is intended.

It can be restricted to the specific directories by having two rules - one for `/qiskit/synthesis` and one for `/test/python/synthesis` (or whatever), with a leading slash.

Reply:
Right, hoping to cover all of `qiskit/synthesis`, `qiskit/transpiler/synthesis`, and `qiskit/quantum_info/synthesis`, plus their associated test directories.